File: home-fall-tracker/javascript/mqtt.js
// The program is using the `mqtt` module
// to publish data to remote MQTT server
var mqtt = require('mqtt');

// The program is using the Node.js built-in `fs` module
// to load the SSL certificates, if needed
var fs = require("fs");

// Publish data to MQTT server
function publish(config, payload) {
  if (!config.MQTT_SERVER || !config.MQTT_CLIENTID) {
    return;
  }

  var data = {d: payload}
  var options = {
    clientId: config.MQTT_CLIENTID,
    username: config.MQTT_USERNAME,
    password: config.MQTT_PASSWORD
  };

  if (config.MQTT_CERT && config.MQTT_KEY) {
    options.cert = fs.readFileSync(config.MQTT_CERT);
    options.key = fs.readFileSync(config.MQTT_KEY);
  }

  console.log("Connecting to MQTT server...");
  var client  = mqtt.connect(config.MQTT_SERVER, options);
  client.on('connect', function () {
    client.publish(config.MQTT_TOPIC, JSON.stringify(data));
    console.log("MQTT message published:", data);
    client.end();
  });
}

exports.log = publish;

exports.increment = function(config) {
  var payload = {counter: 1};
  publish(config, payload);
};

File: alarm-clock/python/mqtt.py
from ssl import PROTOCOL_TLSv1

from paho.mqtt.publish import single as mqtt_publish_single
from paho.mqtt.client import MQTTv311
from simplejson import dumps as serialize_json

# constants

MQTT_SERVER = "MQTT_SERVER"
MQTT_PORT = "MQTT_PORT"
MQTT_CLIENTID = "MQTT_CLIENTID"
MQTT_USERNAME = "MQTT_USERNAME"
MQTT_PASSWORD = "MQTT_PASSWORD"
MQTT_CERT = "MQTT_CERT"
MQTT_KEY = "MQTT_KEY"
MQTT_TOPIC = "MQTT_TOPIC"

def publish_message(config, payload):

    if not { MQTT_SERVER, MQTT_CLIENTID, MQTT_TOPIC } <= set(config): return

    server = config[MQTT_SERVER]
    port = config.get(MQTT_PORT, 1883)

    client_id = config[MQTT_CLIENTID]

    auth = {
        "username": config.get(MQTT_USERNAME),
        "password": config.get(MQTT_PASSWORD)
    }

    tls = {
        "ca_certs": "/etc/ssl/certs/ca-certificates.crt",
        "tls_version": PROTOCOL_TLSv1,
        "certfile": config.get(MQTT_CERT),
        "keyfile": config.get(MQTT_KEY)
    }

    topic = config[MQTT_TOPIC]
    data = serialize_json(payload)

    mqtt_publish_single(topic, payload=data, hostname=server, port=port, client_id=client_id, auth=auth, tls=tls, protocol=MQTTv311)
